pkg/domain/repository/worker.go

```diff
@@ -350,15 +350,15 @@ func (w *worker) stop() {
 		return
 	}
 
+	w.cancel()
+	w.wg.Wait()
+
 	if w.owner != nil {
 		w.owner.Close()
 		w.owner = nil
```

Comment:
Maybe move `w.owner = nil` after `wg.Wait` instead. I think that we should cancel all threads first.

Reply:
I think we need to stop all the threads before it is safe to destroy w.owner.